Streams/StreamWriter.cpp

```diff
@@ -0,0 +1,41 @@
+#include "StreamWriter.h"
+#include "SeekableStreamReader.h"
+#include <vector>
+#include <stdexcept>
+
+void StreamWriter::Write(SeekableStreamReader& seekableStreamReader) {
+	Write(seekableStreamReader, seekableStreamReader.Length() - seekableStreamReader.Position());
+}
+
+void StreamWriter::Write(StreamReader& streamReader, uint64_t writeLength)
+{
+	std::size_t bytesToRead = 0;
+	uint64_t offset = 0;
+	std::vector<char> buffer;
+	buffer.resize(streamPassLength);
+
+	do
+	{
+		bytesToRead = buffer.size();
+
+		// Check if less than a streamPassLength of data remains for writing
+		if (offset + bytesToRead > writeLength) {
+			bytesToRead = static_cast<std::size_t>(writeLength - offset);
+		}
+
+		streamReader.Read(buffer.data(), bytesToRead);
+
+		offset += bytesToRead;
+
+		Write(buffer.data(), bytesToRead);
+	} while (bytesToRead); // End loop when BytesToRead/Write is equal to 0
+}
+
+void StreamWriter::SetStreamPassLength(std::size_t streamPassLength) 
```

Comment:
Err, what?

Reply:
StreamPassLength is the size read from the stream on each pass for writing. The exact same concept as `const std::size_t ARCHIVE_WRITE_SIZE = 0x00020000;`. I thought the following comment explained it reasonably well. Guess not.

```// StreamPassLength is the buffer size used when copying data from a stream reader into the writer. 
// Changing the pass size may affect performance of copying in streams for writing. ```